syntax = "proto3";

package books;

// Serviço para consultas via gRPC
service BookService {
  // Método unário para buscar um livro por nome
  rpc GetBook (BookRequest) returns (BookResponse);
  // Método de streaming para listar todos os livros (opcional)
  rpc ListBooks (Empty) returns (stream BookResponse);
}

// Mensagem para requisição de um livro pela consulta (usando o nome)
message BookRequest {
  string nome = 1;
}

// Mensagem para resposta com os dados do livro
message BookResponse {
  string nome = 1;
  string autor = 2;
  double preco = 3;
}

// Mensagem vazia para o streaming (opcional)
message Empty {}
